### PR TITLE
Support minimizing roomintro/staffintro source code

### PR DESCRIPTION
**Pull request by AnnikaCodes**
Per the suggestion [here](https://www.smogon.com/forums/threads/minimize-roomintro-code.3664984/).

### CODE REVIEW

**Comment by Zarel** [comment]
I commented on the suggestion there. I'd personally prefer it unminimized, and would like to see some kind of poll before approving it.

**Comment by AnnikaCodes** [comment]
> Bump. Did you end up running a poll somewhere?

no, sorry, I'll get on that - I was rather busy this week.

**Comment by Zarel** [comment]
I'm also fine with `<details open>`. I'd rather not keep a PR open forever with no plan to do anything about it.

File: server/chat-commands/room-settings.ts [hunk 1]
```diff
@@ -1137,8 +1137,14 @@ export const commands: ChatCommands = {
 			this.sendReply('|raw|<div class="infobox infobox-limited">' + room.introMessage.replace(/\n/g, '') + '</div>');
 			if (!this.broadcasting && user.can('declare', null, room) && cmd !== 'topic') {
 				this.sendReply('Source:');
```
Comment (Zarel) [inline server/chat-commands/room-settings.ts:1139]:
```suggestion
```

File: server/chat-commands/room-settings.ts [hunk 2]
```diff
@@ -1191,8 +1197,14 @@ export const commands: ChatCommands = {
 			this.sendReply(`|raw|<div class="infobox">${room.staffMessage.replace(/\n/g, ``)}</div>`);
 			if (user.can('ban', null, room) && cmd !== 'stafftopic') {
 				this.sendReply('Source:');
```
Comment (AnnikaCodes) [inline server/chat-commands/room-settings.ts:1199]:
```suggestion
```

File: server/chat-commands/room-settings.ts [hunk 3]
```diff
@@ -1137,8 +1137,14 @@ export const commands: ChatCommands = {
...
-				const code = Chat.escapeHTML(room.introMessage).replace(/\n/g, '<br />');
-				this.sendReplyBox(`<code style="white-space: pre-wrap">/roomintro ${code}</code>`);
+				if (room.introMessage.includes('\n')) {
+					const lines = Chat.escapeHTML(room.introMessage).split('\n');
+					const summary = '/roomintro ' + lines[0];
+					const body = lines.slice(1).join('<br />');
+					this.sendReplyBox(`<details open class="readmore code" style="white-space: pre-wrap"><summary>${summary}</summary>${body}</details>`);
+				} else {
+					this.parse(`/code /roomintro ${room.introMessage}`);
+				}
```
Comment (Zarel) [inline server/chat-commands/room-settings.ts:1146]:
```suggestion
				const code = Chat.escapeHTML(room.introMessage).replace(/\n/g, '<br />');
				this.sendReplyBox(`<details open><summary>Source:</summary><code style="white-space: pre-wrap; display: table; tab-size: 3">/roomintro ${code}</code></details>`);
```

Comment (Zarel) [reply server/chat-commands/room-settings.ts:1146]: (and the same for the other one)

File: server/chat-commands/room-settings.ts [hunk 4]
```diff
@@ -1190,9 +1189,8 @@ export const commands: ChatCommands = {
 			if (!room.staffMessage) return this.sendReply("This room does not have a staff introduction set.");
 			this.sendReply(`|raw|<div class="infobox">${room.staffMessage.replace(/\n/g, ``)}</div>`);
 			if (user.can('ban', null, room) && cmd !== 'stafftopic') {
-				this.sendReply('Source:');
 				const code = Chat.escapeHTML(room.staffMessage).replace(/\n/g, '<br />');
-				this.sendReplyBox(`<code style="white-space: pre-wrap">/staffintro ${code}</code>`);
+				this.sendReplyBox(`<details open><summary>Source:</summary><code style="white-space: pre-wrap; display: table; tab-size: 3">/roomintro ${code}</code></details>`);
```
Comment (Zarel) [inline server/chat-commands/room-settings.ts:1193]:
```suggestion
				this.sendReplyBox(`<details open><summary>Source:</summary><code style="white-space: pre-wrap; display: table; tab-size: 3">/staffintro ${code}</code></details>`);
```

You missed a tiny thing ;)

Comment (AnnikaCodes) [reply server/chat-commands/room-settings.ts]: i'm dumb

**Comment by Zarel** [comment]
Neat, thanks!

**Pull request merged**